### ISSUE TITLE
Game crashes when using stonecutter on certain item when "only craftable" enabled and then the item is no longer craftable.

### ISSUE THREAD

**Issue by orangeisbetter** [Wilyicaro/Legacy-Minecraft#330]
Basically what it says in the title. If you press the button to enable only craftable recipes in the stonecutter interface, and then select one of the recipes and then craft them all, it will consistently (at least in my testing) crash after the last one.

**Comment by orangeisbetter** [comment]
After futher testing, it seems this only occurs in a certain circumstance. I was cutting stone bricks, so I had stone in my inventory. On the 4th slot in the stonecutter, there is the option to craft stone bricks and chiseled stone bricks. Apparently, the bug happened when I crafted the stone bricks, which, upon crafting the first one, an additional option to craft chiseled stone bricks appears (the one using stone bricks instead of stone). This made the cursor stay in the same place, but got bumped onto the other chiseled stone bricks recipe, and I crafted all of those until I ran out of stone. On the last one, the game crashes. I haven't tested this with the crafting table or other blocks that utilize the same interface, but I assume the bug is present there too.

Images of example:
![image](https://github.com/Wilyicaro/Legacy-Minecraft/assets/69521164/e1f0ca47-4895-4dcc-b0c9-408223578fd0)
![image](https://github.com/Wilyicaro/Legacy-Minecraft/assets/69521164/cf600417-7dc6-49db-a996-2a9abf8944f1)
